libs/remix-ui/terminal/src/lib/remix-ui-terminal.tsx

```diff
@@ -0,0 +1,551 @@
+      event.preventDefault()
+      if (autoCompletState.activeSuggestion === 0) {
+        return
+      }
+      setAutoCompleteState(prevState => ({ ...prevState, activeSuggestion: suggestionCount - 1, userInput: Object.keys(autoCompletState.data._options[autoCompletState.activeSuggestion]).toString() }))
+    } else if (event.which === 38 && !autoCompletState.showSuggestions) { // <arrowUp>
+      if (cmdHistory.length - 1 > _cmdIndex) {
+        setCmdIndex(prevState => prevState++)
+      }
+      inputEl.current.innerText = cmdHistory[_cmdIndex]
+      inputEl.current.focus()
+    } else if (event.which === 40 && autoCompletState.showSuggestions) {
+      event.preventDefault()
+      if ((autoCompletState.activeSuggestion + 1) === autoCompletState.data._options.length) {
+        return
+      }
+      setAutoCompleteState(prevState => ({ ...prevState, activeSuggestion: suggestionCount + 1, userInput: Object.keys(autoCompletState.data._options[autoCompletState.activeSuggestion + 1]).toString() }))
+    } else if (event.which === 40 && !autoCompletState.showSuggestions) {
+      if (_cmdIndex > -1) {
+        setCmdIndex(prevState => prevState--)
+      }
+      inputEl.current.innerText = _cmdIndex >= 0 ? cmdHistory[_cmdIndex] : _cmdTemp
+      inputEl.current.focus()
+    } else {
+      setCmdTemp(inputEl.current.innerText)
+    }
+  }
+
+  /* start of mouse events */
+
+  const mousedown = (event: MouseEvent) => {
+    setSeparatorYPosition(event.clientY)
+    const leftRef = document.getElementById('dragId')
+    leftRef.style.backgroundColor = '#007AA6'
+    leftRef.style.border = '2px solid #007AA6'
+    setDragging(true)
+  }
+
+  const onMouseMove: any = (e: MouseEvent) => {
+    e.preventDefault()
+    if (dragging && leftHeight && separatorYPosition) {
+      const newLeftHeight = leftHeight + separatorYPosition - e.clientY
+      setSeparatorYPosition(e.clientY)
+      setLeftHeight(newLeftHeight)
+      event.trigger('resize', [newLeftHeight + 32])
+    }
+  }
+
+  const onMouseUp = () => {
+    const leftRef = document.getElementById('dragId')
+    leftRef.style.backgroundColor = ''
+    leftRef.style.border = ''
+    setDragging(false)
+  }
+
+  /* end of mouse event */
+
+  useEffect(() => {
+    document.addEventListener('mousemove', onMouseMove)
+    document.addEventListener('mouseup', onMouseUp)
+
+    return () => {
+      document.removeEventListener('mousemove', onMouseMove)
+      document.removeEventListener('mouseup', onMouseUp)
+    }
+  })
+
+  React.useEffect(() => {
+    const leftRef = document.getElementById('terminal-view')
+    if (leftRef) {
+      if (!leftHeight) {
+        setLeftHeight(leftRef.offsetHeight)
+        return
+      }
+      leftRef.style.height = `${leftHeight}px`
+    }
+  }, [leftHeight, setLeftHeight, inputEl])
+
+  /* block contents that gets rendered from scriptRunner */
+
+  const _blocksRenderer = (mode) => {
```

Comment:
This is a weird function. It needs refactoring. JSX should be returned not string.